components/date-period-picker/component.jsx

```
@@ -122,11 +142,11 @@ export class DatePeriodPicker extends PureComponent {
 
 		return (
 			<Styled.Container>
-				{datePeriods.map(datePeriod => (
+				{datePeriods.map((datePeriod, index) => (
 					<Styled.DatePeriod
-						key={datePeriod.displayName}
+						key={index}
```

Using indexes as keys can have undesired side effects.
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318
Was `datePeriod.displayName` not guaranteed to be unique? If it is, let's continue to use it. If not, we can *add* the index to the end of the string.

---

Oh, I'd never thought about what would happen if a list needs to be able to change after rendering—that _is_ a habit I should get out of. Thanks for the info!

The only situation I can see in which `datePeriod.displayName` would _not_ be unique is if the dev made a mistake, because seeing two identically labeled buttons has got to be confusing to a user (and unnecessary here).

---

>The only situation I can see in which datePeriod.displayName would not be unique is if the dev made a mistake, because seeing two identically labeled buttons has got to be confusing to a user (and unnecessary here).

@RobertBolender, @tywmick, to this end, are you opposed to adding some validation to this component that the consumer did *not* provide duplicate display names? I'd rather us fail fast for a developer with a clear error message than them run into unexpected behavior because of duplicate `key`s.

---

Ooh! Yeah, one thing I meant to say in my previous comment but forgot was that one advantage of using the display name as the key would be that React would send a key warning if someone used duplicate display names. But you're right, @rmjohnson, a more descriptive error message would be much more useful.

What would be best here, a console warning, console error, or throwing an error in development?

---

In this case, the duplicate keys will cause a clear (enough) warning message for us.
I'd be open to additional validation, but would be fine considering that out of scope for this PR and merging this one as-is.

My one hesitation *away* from failing fast in this case is my hypothetical case where a runtime-generated list *doesn't* fail for the developer but fails for the user, where we might end up showing them confusing results that *might* "just work" if we hadn't interrupted their experience with an error message. It's hard to evaluate this scenario without something more concrete.

How would you feel about a filtering the list for duplicates, keeping only the first for each display string, and logging a warning/error for any duplicates? (Not blocking for this PR, it can be a follow-up)

---

> How would you feel about a filtering the list for duplicates, keeping only the first for each display string, and logging a warning/error for any duplicates? (Not blocking for this PR, it can be a follow-up)

I like it! 👌🏼